src/Language/Haskell/TH/DeepStrict.hs

```diff
@@ -144,7 +156,7 @@ data Levity = Lifted | Unlifted
 -- We use these to annotate types with deep strictness overrides.
 -- Types that have fields labelled as 'Language.Haskell.TH.DeepStrict.Strict' require those types to be deep strict.
 -- Types that have fields labelled as 'Language.Haskell.TH.DeepStrict.Lazy' will never be deep strict, but this can be helpful for nicer messages.
-data Strictness = Strict | Lazy
+data Strictness = UnliftedStrictness | Strict | Lazy
```

Comment:
```suggestion
data ExpectStrictness = 
  ExpectUnlifted 
  | ExpectStrict 
  | ExpectAnything
```
Rereading this stuff, I think this type is quite confusing. Maybe we could rename it to be called something like this, since really this is an expectation. The lazy one means we don't care.

Reply:
Maybe not a job for this MR though